### PR TITLE
Update Category For ID

### PR DESCRIPTION
**Pull request by walterio123**
**Summary**

- Creation CategoryEntityDtoFull.
- Modify CategoryMapper for CategoryEntityFullDto.
- Creation of method in CategoryService to update the Category entity.
- Creation of the route in Controller for updating the Category enitity.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/service/CategoryService.java [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 import java.util.List;
-
+@RequiredArgsConstructor
 @Service
 public class CategoryService {
 
-    private CategoryMapper categoryMapper;
+    private final CategoryMapper categoryMapper;
     @Autowired
     private CategoryEntityRepository categoryEntityRepository;
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/service/CategoryService.java:21]: should we refactor this repository to CategoryRepository? I think is simpler

File: src/main/java/com/alkemy/ong/controller/CategoryController.java [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/category")
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/controller/CategoryController.java:13]: I am not sure that the ticket suggest this route /category, in stead strait /categories

File: src/main/java/com/alkemy/ong/entity/CategoryEntity.java [hunk 3]
```diff
@@ -11,7 +11,6 @@
 @Entity
 @Getter @Setter @ToString @RequiredArgsConstructor
 @SQLDelete(sql="UPDATE categorys SET softDelete = true WHERE id=?")
```
Comment (anavduo) [inline src/main/java/com/alkemy/ong/entity/CategoryEntity.java:13]: I suggest the table's name should be categories, that is the plural(here is says categorys). If we refactor the table's name, we need to take into account to drop a create the DB afterwards

File: src/main/java/com/alkemy/ong/service/CategoryService.java (hunk 1 above)
Comment (walterio123) [reply src/main/java/com/alkemy/ong/service/CategoryService.java:21]: I think the same, I will raise it in the next meeting

File: src/main/java/com/alkemy/ong/entity/CategoryEntity.java (hunk 3 above)
Comment (walterio123) [reply src/main/java/com/alkemy/ong/entity/CategoryEntity.java:13]: We will agree on it, and we will modify it if it is so.I agree with your way of seeing it.

File: src/main/java/com/alkemy/ong/controller/CategoryController.java (hunk 2 above)
Comment (walterio123) [reply src/main/java/com/alkemy/ong/controller/CategoryController.java:13]: The ticket requests this route: PUT /categories/:id if I'm not correct, I'll correct it.

**Review by juan1977lopez — APPROVED** [review]
Approved

**Review by juan1977lopez — APPROVED** [review]
Approved

**Pull request merged**